File: Day285/findTheOriginalArrayOfPrefixXOR.go
package main

import (
	"fmt"
)

func findArray(pref []int) []int {
	prev := pref[0]
    for i:=1; i<len(pref); i++ {
        temp := pref[i]
        pref[i] ^= prev
        prev = temp
    }
    return pref

}
func main() {
	arr1 := []int{5,2,0,3,1}
	arr2 := []int{13}
	fmt.Println(findArray(arr1))
	fmt.Println(findArray(arr2))
}

File: Day285/findTheOriginalArrayOfPrefixXOR.java
package Day285;

import java.util.Arrays;

class findTheOriginalArrayOfPrefixXOR {
    public static int[] findArray(int pref[]) {
        for (int i= pref.length- 1; i>0; i--) {
            pref[i] ^= pref[i-1];
        }
        return pref;
    }
    public static void main(String[] args) {
        int arr1[] = {5,2,0,3,1};
        int arr2[] = {13};
        System.out.println(Arrays.toString(findArray(arr1)));
        System.out.println(Arrays.toString(findArray(arr2)));
    }
}